src/base.js

```diff
@@ -13,8 +13,225 @@
+            document.webkitFullscreenElement ||
+            document.mozFullScreenElement ||
+            document.msFullscreenElement;
+
+        self.isPresenting = fullscreenElement == layer.source;
+        self.fireVRDisplayPresentChange_();
+        if (self.isPresenting) {
+          if (screen.orientation && screen.orientation.lock)
+            screen.orientation.lock("landscape-primary");
+          self.waitingForPresent_ = false;
+          self.beginPresent_();
+          resolve();
+        } else {
+          if (screen.orientation && screen.orientation.unlock)
+            screen.orientation.unlock();
+          self.endPresent_();
+          self.removeFullscreenListeners_();
+        }
+      }
+      function onFullscreenError() {
+        if (!self.waitingForPresent_)
+          return;
+
+        self.removeFullscreenListeners_();
+
+        self.waitingForPresent_ = false;
+        self.isPresenting = false;
+
+        reject(new Error("Unable to present."));
+      }
+
+      self.addFullscreenListeners_(layer.source,
+          onFullscreenChange, onFullscreenError);
+
+      self.waitingForPresent_ = true;
+      // TODO: Spec says that requestFullscreen returns a promise, but that
+      // doesn't seem to be implemented anywhere just yet.
+      if (layer.source.requestFullscreen)
+        layer.source.requestFullscreen();
+      else if (layer.source.webkitRequestFullscreen)
+        layer.source.webkitRequestFullscreen();
+      else if (layer.source.mozRequestFullScreen)
+        layer.source.mozRequestFullScreen();
+      else if (layer.source.msRequestFullscreen)
+        layer.source.msRequestFullscreen();
+      else if (Util.isIOS()) {
+        // *sigh* Just fake it.
+        self.isPresenting = true;
+        self.fireVRDisplayPresentChange_();
+        self.beginPresent_();
+        resolve();
+      } else
+        self.waitingForPresent_ = false;
+    }
+
+    if (!self.waitingForPresent_) {
+      document.exitFullscreen();
+      reject(new Error("Unable to present."));
+    }
+  });
+};
+
+VRDisplay.prototype.exitPresent = function() {
+  var wasPresenting = this.isPresenting;
+  this.isPresenting = false;
+  this.layer_ = null;
+
+  return new Promise(function(resolve, reject) {
+    if (wasPresenting) {
+      // TODO: Spec says that exitFullscreen returns a promise, but that
+      // doesn't seem to be implemented anywhere just yet.
+      if (document.exitFullscreen)
+        document.exitFullscreen();
+      else if (document.webkitExitFullscreen)
+        document.webkitExitFullscreen();
+      else if (document.mozCancelFullScreen)
+        document.mozCancelFullScreen();
+      else if (document.msExitFullscreen)
+        document.msExitFullscreen();
+
+      resolve();
+    } else {
+      reject(new Error("Was not presenting to VRDisplay."));
+    }
+  });
+};
+
+VRDisplay.prototype.getLayers = function() {
+  if (this.layer_) {
+    return [this.layer_];
+  }
+  return null;
```

Comment:
Spec currently lists this function as returning `sequence<VRLayer>?`, I figured that since it's nullable that implies that `null` is the default return value when nothing is set. If that's not the intended behavior we should change the spec to make the return value non-nullable. Either way the spec should have text explaining what the expected return values are.